### DISCUSSION TITLE
can't add new columns to BB internal USERS table

### DISCUSSION THREAD

**Discussion by JDiego199** [Budibase/budibase#11957] [Help] [answered]
A "+" is supposed to appear at the end of the last column to add new ones and nothing appears

![image](https://github.com/Budibase/budibase/assets/68968967/498623d8-bdb5-488e-a6d1-e6c42807b0b6)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @JDiego199,

The new release has removed this functionality. If you are wanting to use relationships for the users table you can use the new `User` column type as highlighted in the latest release announcement [here](https://budibase.com/blog/updates/2023/september/).

I hope this helps!

**Reply by MikeYates** [reply]
Thanks for informing, this update made me to redo all my app functions.  Its nice, but this kind of update is scary as my app just stop functioning. I literally went lightheaded when I was about to put my app into production today.

**Reply by ConorWebb96** [reply]
Sorry to hear you had a lot of trouble. Hopefully this is an easier way for you to create relationships between your data and users going forward. 🙌

**Reply by ConorWebb96** [reply]
Hey @MikeYates,

Can I ask what exactly the issues you experienced where? 

Existing columns created previously should have remained and shouldn't have been affected. Previously created relationship columns also should have been fine after the update.

Thanks in advance.

**Reply by MikeYates** [reply]
@ConorWebb96 

All of my data providers in my app are filtered to specific user, so in production I have to assign a specific code for every new user. Previously, through the user database I would create a column to indentify the user with a specific code (eg: "user_id: 1234")

And in all my database table data has a column called `user_id`, so this functions whenever that user saves a row only themself will see their own row even though its populated from various users. In all the data provider are filtered with a `user_id` column with value binded by curent user `user_id`.

With this new update i just have to redo all the table column with the new `user column` and update all the data provider filters with the `user column` although im running into a problem where the filter type for `user column` shows up blank, so I cannot filter it. There is no "Equals, Like, Starts with, etc" filters.

What should I do?

UPDATE:
I submitted a bug and I believe its confirmed to be a bug and been set on high priority. Phew, I thought it was intentional.

[/issues/11958]

**Reply by ConorWebb96** [reply]
Hey @MikeYates,

Thank you for creating the issue. Yes, this does seem to be related to the users change. One of our engineers is currently investigating and looking at a fix for this.

Thanks again for raising this.

**Reply by MikeYates** [reply]
@ConorWebb96 Do you have a rough estimate when this bug might be patched or how long does it generally take? Sorry I don't mean to be pushy, I was just supposed to set my app into production yesterday (during the update) and kind of depend on that bug for my whole app to function. What a coincidence aye.

**Reply by ConorWebb96** [reply]
Hey @MikeYates,

I believe this has been fixed with the latest [release](https://github.com/Budibase/budibase/issues/11958).

Hopefully this solves it.
